pisa/stages/xsec/genie_sys.py

```diff
@@ -23,32 +23,32 @@ class genie_sys(Stage): # pylint: disable=invalid-name
     ----------
     params
         Must contain ::
-
-            Genie_Ma_QE : quantity (dimensionless)
-            Genie_Ma_RES : quantity (dimensionless)
+        
+        parameters specified in interactions (dimensionless)
 
     Notes
     -----
-    Requires the events have the following keys ::
+    Requires the events have the following keys for each included interaction ::
 
-        linear_fit_maccqe
-            Genie CC quasi elastic linear coefficient
-        quad_fit_maccqe
-            Genie CC quasi elastic quadratic coefficient
-        linear_fit_maccres
-            Genie CC resonance linear coefficient
-        quad_fit_maccres
-            Genie CC resonance quadratic coefficient
+        linear_fit_{name}
+            Genie linear coefficient for interaction {name}
+        quad_fit_{name}
+            Genie quadratic coefficient for interaction {name}
 
     """
     def __init__(
         self,
+        interactions="Genie_Ma_QE Genie_Ma_RES",
+        names="maccqe maccres",
         **std_kwargs,
     ):
-        expected_params = (
-            'Genie_Ma_QE',
-            'Genie_Ma_RES',
-        )
+        interactions = tuple(interactions.split(' '))
```

Comment:
It's just a suggestion, it might be convenient for later usage to split the string by not only ' ', but also if ',' or ', ' are used. I think re.split() can do that?
Anyway, only a suggestion